### PR TITLE
Cleanup Jest Test Output (part 2)

### PR DESCRIPTION
**Pull request by jmchilton**
Redo and expansion of #19164.

## How to test the changes?
(Select all options that apply)
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/tests/jest/jest.setup.js [hunk 1]
```diff
@@ -4,10 +4,25 @@ import "fake-indexeddb/auto";
...
 
+// if we do jest.mock(module) where modules imports Vue, it will create a clean Vue context
+// that ignores the above tips. So pre-emptively mock out Vue in these contexts with the following
+// block of code. GridList.test.js is a good example of this, when ran (yarn test GridList.test.js)
+// it will bypass the attempt to set the Vue config values above and print the messages anyway without
+// this block.
+jest.mock("vue", () => {
+    const Vue = jest.requireActual("vue"); // Import the actual Vue instance
+    Vue.config.productionTip = false; // Disable production tip
+    Vue.config.devtools = false; // Disable production tip
```
Comment (dannon) [inline client/tests/jest/jest.setup.js]:
```suggestion
    Vue.config.devtools = false; // Disable vue devtools
```

Comment (jmchilton) [reply client/tests/jest/jest.setup.js]: Thanks!

Comment (dannon) [reply client/tests/jest/jest.setup.js]: Thank you!  This is fantastic cleanup for the tests.

**Review by davelopez — APPROVED** [review]
Thank you so much for all the cleaning!

I've rerun the client tests that were failing and they are passing now:

https://github.com/galaxyproject/galaxy/actions/runs/11958564319/job/33338362283?pr=19178

Likely because of the random nature of:
```
A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --detectOpenHandles to find leaks. Active timers can also cause this, ensure that .unref() was called on them.
```
There are many more things in the test output that can be fixed, but this is already a HUGE improvement! We can continue iterating on the cleanup in smaller chunks.
Thank you!

**Pull request merged**